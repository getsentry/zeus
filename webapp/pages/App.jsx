import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {authSession} from '../actions/auth';
import {loadRepos} from '../actions/repos';

import AsyncComponent from '../components/AsyncComponent';
import ErrorBoundary from '../components/ErrorBoundary';
import Indicators from '../components/Indicators';
import PageLoadingIndicator from '../components/PageLoadingIndicator';

export class RepositoryContext extends AsyncComponent {
  static propTypes = {
    repoList: PropTypes.arrayOf(PropTypes.object),
    ...AsyncComponent.propTypes
  };

  static childContextTypes = {
    ...RepositoryContext.propTypes
  };

  getChildContext() {
    return {
      repoList: this.props.repoList
    };
  }

  fetchData() {
    return new Promise((resolve, reject) => {
      this.props.loadRepos();
      return resolve();
    });
  }
}

const AuthedContext = connect(
  function(state) {
    return {
      repoList: state.repos.items,
      loading: !state.repos.loaded
    };
  },
  {loadRepos}
)(RepositoryContext);

class App extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool,
    user: PropTypes.object,
    authSession: PropTypes.func.isRequired
  };

  componentWillMount() {
    this.props.authSession();
  }

  getTitle() {
    return 'Zeus';
  }

  render() {
    return (
      <div>
        <Indicators />
        <ErrorBoundary>
          {!this.props.isAuthenticated === null ? (
            <PageLoadingIndicator />
          ) : (
            <AuthedContext>{this.props.children}</AuthedContext>
          )}
        </ErrorBoundary>
      </div>
    );
  }
}

export default connect(
  ({auth}) => ({
    user: auth.user,
    isAuthenticated: auth.isAuthenticated
  }),
  {authSession}
)(App);
