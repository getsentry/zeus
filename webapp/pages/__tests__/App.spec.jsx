import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';

import App from '../App';

const mockStore = configureStore();

describe('App', () => {
  it('renders without crashing', () => {
    let store = mockStore({
      auth: {}
    });
    const tree = shallow(<App store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
