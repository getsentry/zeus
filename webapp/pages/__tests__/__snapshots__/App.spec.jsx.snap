// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App
  authSession={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
